manifests/params.pp

```diff
@@ -18,28 +18,28 @@
   # strict variables wasn't added until 3.5.0, so this should be fine.
   if ! $::settings::strict_variables {
     $xfacts = {
-      'lsbdistid'           => $::lsbdistid,
-      'lsbdistcodename'     => $::lsbdistcodename,
-      'lsbmajdistrelease'   => $::lsbmajdistrelease,
-      'lsbdistrelease'      => $::lsbdistrelease,
+      'lsbdistid'           => $facts['os']['distro']['id'],
+      'lsbdistcodename'     => $facts['os']['distro']['codename'],
+      'lsbmajdistrelease'   => $facts['os']['distro']['release']['major'],
+      'lsbdistrelease'      => $facts['os']['distro']['release']['full'],
     }
   } else {
     # Strict variables facts lookup compatibility
     $xfacts = {
       'lsbdistid' => defined('$lsbdistid') ? {
-        true    => $::lsbdistid,
+        true    => $facts['os']['distro']['id'],
         default => undef,
       },
       'lsbdistcodename' => defined('$lsbdistcodename') ? {
-        true    => $::lsbdistcodename,
+        true    => $facts['os']['distro']['codename'],
         default => undef,
       },
       'lsbmajdistrelease' => defined('$lsbmajdistrelease') ? {
-        true    => $::lsbmajdistrelease,
+        true    => $facts['os']['distro']['release']['major'],
         default => undef,
       },
       'lsbdistrelease' => defined('$lsbdistrelease') ? {
-        true    => $::lsbdistrelease,
+        true    => $facts['os']['distro']['release']['full'],
         default => undef,
       },
     }
```

Comment:
This approach doesn't work with structured facts. If `lsb-release` is not installed, `$facts['os']['distro']` is not defined and you'll get an error that you can't call `[]` on `Undef`. I'd suggest to use `fact()` from stdlib which does the right thing. This also removes the need for the special case.

Reply:
In another module, we use ```dig()``` to check if ```$facts['os']['distro']``` exists.

Reply:
But [fact()](https://forge.puppet.com/puppetlabs/stdlib/reference#fact) was exactly designed for this purpose.

Reply:
I am just saying: 
* https://github.com/voxpupuli/puppet-telegraf/pull/146

Reply:
at https://voxpupuli.org/docs/reviewing_pr/ , we agreed on using `fact()`:
> Are facts used? They should only be accessed via `$facts[]` or `fact()` function from stdlib, but not topscope variables

so I would go with `fact()` in this particular usecase, or uppdate our styleguide.

Reply:
I just found another module using ```dig```:
* https://github.com/voxpupuli/puppet-nginx/pull/1376

Reply:
I know that we will have this discussion regularly, but sometimes the code name is needed. For example, because it is part of the URL to a repository.

Reply:
That's different because it's Ruby, not Puppet.

Reply:
Just to be explicit, I think this whole block can be replaced with:
```puppet
$xfacts = {
  'lsbdistid'         => fact('os.distro.id'),
  'lsbdistcodename'   => fact('os.distro.codename'),
  'lsbmajdistrelease' => fact('os.distro.release.major'),
  'lsbdistrelease'    => fact('os.distro.release.full'),
}
```

Reply:
Not according to my local tests.

Reply:
And we can probably get rid of a lot anyway: https://github.com/voxpupuli/puppet-unattended_upgrades/pull/164 is a start.

Reply:
https://github.com/voxpupuli/puppet-unattended_upgrades/pull/165 would take it a step further. I'd suggest to look at those 2 PRs and then rebase modulesync based on that. In my experience it's better to first simplify and modernize.